sound/soc/intel/boards/sof_sdw.c

```diff
@@ -906,6 +921,10 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 	if (codec_info_list[codec_index].ignore_pch_dmic)
 		*ignore_pch_dmic = true;
 
+	if (codec_info_list[codec_index].codec_type == SOF_SDW_CODEC_TYPE_AMP &&
+	    *link_id < SDW_AMP_DAI_ID)
+		*link_id = SDW_AMP_DAI_ID;
+
```

Comment:
I guess this is ok, but  a little difficult to follow. So if link_id is larger than SDW_AMP_DAI_ID, then it's not fixed anymore. Should this be an error?

Reply:
what happens otherwise? is it legal? or are you leaving that at the mercy of topology changes?

Reply:
Thanks @bardliao . I wonder should this be explained either in the commit message or a code comment. This is something anyone maintaining the code in the future has to understand.

Reply:
> Thanks @bardliao . I wonder should this be explained either in the commit message or a code comment. This is something anyone maintaining the code in the future has to understand.

Code comment is added.